### ISSUE TITLE
Stop

### ISSUE THREAD

**Issue by matzZz** [nicoh88/homebridge-xiaomi-roborock-vacuum#73]
Hi,
when I stop the vacuum with the HomeKit switch the robot stops and don't go to the dock.
is there a solution that due robot go to the dock?

**Comment by afharo** [comment]
Hi @matzZz, do you have the Pause switch enabled?

**Comment by matzZz** [comment]
I tried with pause switch enabled and disabled.
But when I switch off the vacuum stops in the room and don’t go back to the dock.

**Comment by afharo** [comment]
It happens to me as well with the Pause switch enabled (haven't tried it without it). Ever since iOS13, apparently the combined view gets both switches mixed up and I can hear the robot saying "Paused". If I see the individual accessories and turn the robot off in that view, it successfully stops and goes back to the dock. I have the `roborock.vacuum.s5` model though.

**Comment by matzZz** [comment]
I has the Xiaomi 1s.
Now I tried a little. When I switch off the fan-switch says the robot „paused“, when i paused with the pause-switch and then switch off the fan-switch the robot goes back to the dock...?!

**Comment by afharo** [comment]
So same issue as I have... I’ll try to find out a way to workaround the iOS13 new behaviour

**Comment by rellinger** [comment]
+1 same issue as matzZz

**Comment by pelo76** [comment]
Hi, I have the same issue with Roborock S6... Would it be possible to add a "Go Home" Button?

Great work so far, thanks a lot

**Comment by afharo** [comment]
Sorry for the wait! This is my next planned task.

**Comment by afharo** [comment]
I’ve been playing around with the current behaviour in iOS13 and it looks like, if the speed is dropped to 0%, it goes back to the dock.

Can you please confirm on your end? And, if not, what versions are not doing so?

**Comment by afharo** [comment]
I've tried converting the Pause switch to the Fan RotationDirection. 
It looks like:
![image](https://user-images.githubusercontent.com/5469006/73676327-1cd10100-46ac-11ea-8f73-508c1c582a22.png)
This only confirms that the ON-OFF _switch_ in a fan in the new iOS13 is actually presented as 0% speed or the actual fan speed. So, in order to "stop" the cleaner and call it to go to the dock, we simply need to slide the _speedometer_ down to 0%.
Another option is to see the options of the device
![image](https://user-images.githubusercontent.com/5469006/73676741-d039f580-46ac-11ea-84ca-b7e30b2d90e0.png)
And tap on "Accessories", leading to this view:
![image](https://user-images.githubusercontent.com/5469006/73676789-e3e55c00-46ac-11ea-89da-8f9489773127.png)
Then tapping on the actual "fan" will stop the robot and send it to the dock for charging.

**Comment by serbasan** [comment]
Hi. Zero fan, stop the vacuum in my case. I can't get it back to the charging base.

**Comment by Tim2309** [comment]
The bug still exist with the S6
if you set the speed to 0% the robot stop and say "spot cleaning"
if you first pause the robot and then set to speed 0% to robot is going back to the dock

**Comment by afharo** [comment]
Please, follow the updates on the issue @Tim2309 just created and the linked PR #106

**Comment by serbasan** [comment]
Thanks, that way it works

**Comment by nicoh88** [comment]
Successfully published homebridge-xiaomi-roborock-vacuum@0.6.8

```
sudo su -
npm -g install homebridge-xiaomi-roborock-vacuum@latest
```
